src/wsdiscoveryclient.cpp

```diff
@@ -73,15 +75,20 @@ void WSDiscoveryClient::sendProbe(const QList<KDQName> &typeList, const QList<QU
     KDSoapMessageAddressingProperties addressing;
     addressing.setAddressingNamespace(KDSoapMessageAddressingProperties::Addressing200408);
     addressing.setAction(QStringLiteral("http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe"));
-    addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString(QUuid::WithoutBraces));
+    addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString().remove("{").remove("}"));
     addressing.setDestination(QStringLiteral("urn:schemas-xmlsoap-org:ws:2005:04:discovery"));
-    addressing.setReplyEndpointAddress(KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::Anonymous));
+    addressing.setReplyEndpointAddress(KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::Anonymous,
+                                                                                                    KDSoapMessageAddressingProperties::Addressing200408));
     message.setMessageAddressingProperties(addressing);
 
     auto rc = m_soapUdpClient->sendMessage(message, KDSoapHeaders(), DISCOVERY_ADDRESS_IPV4, DISCOVERY_PORT);
-    Q_ASSERT(rc);
+    if (!rc) {
+        qWarning() << "sendProbe()->KDSoapUdpClient::sendMessage() returned false for" << DISCOVERY_ADDRESS_IPV4 << "on port" << DISCOVERY_PORT;
+    }
```

Comment:
Again, I want a hard error when sendMessage() fails. Maybe you can persuade me to do `Q_ASSERT(rc1 || rc2);`...